### DISCUSSION TITLE
UMD documentation on component usage

### DISCUSSION THREAD

**Discussion by murugappanrm** [quasarframework/quasar#10769] [General - Components / Directives / etc] [answered]
I am trying to use the UMD standalone option provided by Quasar. The absolute disgusting part of this good framework is that it provides hopeless documentation on a feature listed - UMD. 

I tried like hell and except for a few vue components i could not use things like layout and the sorts. Each individual are is documented for CLI but nothing is mentioned on UMD. Really terrible. 

Can anyone point me to a document which provides details how to use the UMD? I cant even get a YouTube video on this.

**Comment by metalsadman** [comment]
All example codepens are using UMD, if https://quasar.dev/start/umd isn't enough.
![image](https://user-images.githubusercontent.com/31038838/134288496-b534c03e-bff7-4bbf-afd4-454ab39a93cb.png)

**Reply by murugappanrm** [reply]
Thank you so much. How to use the codepen was not detailed.  Suggest this be made in the https://quasar.dev/start/umd page. That way it would be good. Sad, i lost about 4 days knocking my head into the howto.

**Comment by cyfung1031** [comment, marked as answer]
Just fork all the CodePen examples from the official docs.

**Reply by murugappanrm** [reply]
Hi, Thank you so much.

**Reply by murugappanrm** [reply]
@cyfung1031, I do not know how to fork all the codepen examples.

**Reply by pdanpdan** [reply]
Click on the open codepen button that is present in the top bar of every example

**Comment by murugappanrm** [comment]
Hi, with the wonderful help from you guys, i managed to get the layout working. I have become an ardent fan for Vuejs and Quasar. 

After some learning and testing, i found myself in a  bit of fix. This is my problem, maybe others too, as i was trying to mimic what i did using the cli version ( a todo example):

(1) How do i inject a page into the layout as in CLI? I created a the layout and it worked fine. Now i want to create a page separately and then inject it into the layout. I tried so many ways and could not do it. 
(2) How do i create the menu items in site drawer as in the CLI which uses "EssentialLink" components?
(3) How do i use the "template" tag in UMD?

There is much help in the components for UMD by referring to the codepen version but it will be great if there are indications in other areas where it can be stated that a particular feature is or is not applicable under UMD main feature. 

Basically, what trying to do is insert Quasar into our existing codeigniter and HTML based web apps.  Help is much appreciated. I am taking baby steps here.

**Comment by murugappanrm** [comment]
Looks the help end here, i guess. Sigh!

**Reply by metalsadman** [reply]
first read VUE DOCS, their examples are mostly for UMD, from creating reusable components and etc...

**Reply by murugappanrm** [reply]
Thank you, I guess you wrote a funny answer initially and then changed it later. Unfortunately, both message were dispatched.  Never mind.
